Add frontend metrics to default metrics configuration

namannandan opened the pull request:
## Description

Add frontend metrics collected by [system_metrics.py](https://github.com/pytorch/serve/blob/master/ts/metrics/system_metrics.py) to [default metrics configuration file](https://github.com/pytorch/serve/blob/master/ts/configs/metrics.yaml)

Fixes #2133

lxning reviewed (commented):
Please search "new Metric" in frontend java code, add all the metrics generated from the metrics' objects.

lxning commented on ts/configs/metrics.yaml:
```diff
@@ -11,13 +12,46 @@ ts_metrics:
     - name: GaugeTsMetricExample
       unit: ms
       dimensions: [*model_name, *level]
```
can you point out the code using this metrics?

lxning commented on ts/configs/metrics.yaml:
```diff
@@ -11,13 +12,46 @@ ts_metrics:
+      unit: percent
+      dimensions: [*level, *device_id]
   histogram:
     - name: HistogramTsMetricExample
       unit: ms
       dimensions: [*model_name, *level]
```
can you point out the code using this metrics?

lxning commented on ts/configs/metrics.yaml:
```diff
@@ -11,13 +12,46 @@ ts_metrics:
 
 model_metrics:
   counter:
+    - name: CounterModelMetricNameExample
+      unit: ms
+      dimensions: [*model_name, *level]
```
can you point out the code using this metrics?

lxning commented:
@namannandan Please 
- first clean up unused metric in the existing metrics.yaml file.
- second search all existing metrics in frontend and backend code, add them into ts_metrics or model_metrics in the metrics.yaml.

namannandan commented:
Addressed review comments.
Skipped [WorkerLoadTime](https://github.com/pytorch/serve/blob/86d440041b663961c71a6262fe648111d85b27d8/frontend/server/src/main/java/org/pytorch/serve/wlm/WorkerThread.java#L163) metric since the metric name is not fixed and is [determined at runtime](https://github.com/pytorch/serve/blob/86d440041b663961c71a6262fe648111d85b27d8/frontend/server/src/main/java/org/pytorch/serve/wlm/WorkerThread.java#L164).

lxning commented on ts/configs/metrics.yaml:
```diff
@@ -1,34 +1,62 @@
+      unit: percent
+      dimensions: [*level, *device_id]
 
 model_metrics:
```
Add comment in the file.
dimension "hostname" is automatically added for model metrics at backend.

lxning commented on ts/configs/metrics.yaml:
```diff
@@ -1,34 +1,62 @@
   - &level "Level"
+  - &device_id "device_id"
 
 ts_metrics:
```
missing hostname in all ts_metrics

namannandan replied:
Added hostname dimensioin

lxning reviewed (commented):
the following parts are missing:

- mode (check design doc)
- metrics defined in [prometheus](https://github.com/pytorch/serve/blob/master/frontend/server/src/main/java/org/pytorch/serve/metrics/format/prometheous/PrometheusMetricManager.java)
- WorkerLoadTime metric

namannandan commented:
Addressed review comments:
- added `mode` configuration
- added metrics defined in [prometheus](https://github.com/pytorch/serve/blob/master/frontend/server/src/main/java/org/pytorch/serve/metrics/format/prometheous/PrometheusMetricManager.java)
- added `WorkerLoadTime` metric. Since the metric name is determined at [runtime](https://github.com/pytorch/serve/blob/86d440041b663961c71a6262fe648111d85b27d8/frontend/server/src/main/java/org/pytorch/serve/wlm/WorkerThread.java#L164), included a dimension called `WorkerName` for this metric.

lxning commented on ts/configs/metrics.yaml:
```diff
@@ -1,34 +1,80 @@
+    - name: Requests5XX
+      unit: Count
+      dimensions: [*level, *hostname]
+    - name: InferenceRequestsTotal
+      unit: Count
+      dimensions: [*model_name, *model_version, *hostname]
+    - name: CumulativeInferenceLatencyMicroseconds
+      unit: Microseconds
+      dimensions: [*model_name, *model_version, *hostname]
+    - name: CumulativeQueueLatencyMicroseconds
+      unit: Microseconds
+      dimensions: [*model_name, *model_version, *hostname]
```
Please use original prometheus metric name, not help.
